generic_config_updater/services_validator.py

```diff
@@ -17,8 +19,35 @@ def set_verbose(verbose=False):
 
 def _service_restart(svc_name):
     rc = os.system(f"systemctl restart {svc_name}")
-    logger.log(logger.LOG_PRIORITY_NOTICE,
-            f"Restarted {svc_name}", print_to_console)
+    if rc != 0:
+        # This failure is likely due to too many restarts
+        # Which is the reflection of patch sorter not grouping
+        # updates for keys effectively.
+        # Once the patch sorter is tuned, the following
+        # code would not be executed.
+        #
+        rc = os.system(f"systemctl reset-failed {svc_name}")
+        logger.log(logger.LOG_PRIORITY_ERROR, 
+                f"Service has been reset. rc={rc} Pause for {PAUSE_AFTER_FAILED_RESTART}",
+                print_to_console)
+
+        # Even with successful reset-failed, a pause is required, else the
+        # immediate restart was found to fail in manual tests.
+        # Hence add a pause.
+        # This pause would help even if the earlier reset-failed fails.
+        # Hence we ignore return code from reset-failed.
+        #
+        os.system(f"sleep {PAUSE_AFTER_FAILED_RESTART}")
```

Comment:
Curios why do we need to wait, I tried manually and it seems to be ok to not wait.

Steps:
* create file `script` with content:
```print_and_exec()
{
    # get time 
    timestamp=$(date +"%T.%N")
    # print command
    echo [$timestamp] RUNNING: $@ at 

    # exec command
    $@
    exit_code=$?
}

SERVICE_NAME=$1
print_and_exec systemctl status $SERVICE_NAME
for (( ; ; )) # infinite loop
do
    print_and_exec sudo systemctl restart $SERVICE_NAME
    if [ $exit_code -eq 1 ]; then
        break
    fi
done
print_and_exec systemctl status $SERVICE_NAME
print_and_exec sudo systemctl reset-failed $SERVICE_NAME
print_and_exec systemctl status $SERVICE_NAME
print_and_exec sudo systemctl restart $SERVICE_NAME
print_and_exec systemctl status $SERVICE_NAME
```
* Run `bash script dhcp_relay`:
```
admin@vlab-01:~$ bash script dhcp_relay
[20:58:07.138839551] RUNNING: systemctl status dhcp_relay at
● dhcp_relay.service - dhcp_relay container
     Loaded: loaded (/lib/systemd/system/dhcp_relay.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/dhcp_relay.service.d
             └─auto_restart.conf
     Active: active (running) since Tue 2022-01-18 20:57:53 UTC; 13s ago
    Process: 288332 ExecStartPre=/usr/local/bin/dhcp_relay.sh start (code=exited, status=0/SUCCESS)
   Main PID: 288422 (dhcp_relay.sh)
      Tasks: 3 (limit: 3303)
     Memory: 18.0M
     CGroup: /system.slice/dhcp_relay.service
             ├─288422 /bin/bash /usr/local/bin/dhcp_relay.sh wait
             ├─288444 /bin/bash /usr/bin/dhcp_relay.sh wait
             └─288445 python3 /usr/local/bin/container wait dhcp_relay
[20:58:07.183838665] RUNNING: sudo systemctl restart dhcp_relay at
[20:58:14.789788540] RUNNING: sudo systemctl restart dhcp_relay at
[20:58:20.890107725] RUNNING: sudo systemctl restart dhcp_relay at
Job for dhcp_relay.service failed.
See "systemctl status dhcp_relay.service" and "journalctl -xe" for details.
[20:58:24.296103250] RUNNING: systemctl status dhcp_relay at
● dhcp_relay.service - dhcp_relay container
     Loaded: loaded (/lib/systemd/system/dhcp_relay.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/dhcp_relay.service.d
             └─auto_restart.conf
     Active: failed (Result: start-limit-hit) since Tue 2022-01-18 20:58:24 UTC; 24ms ago
    Process: 288791 ExecStartPre=/usr/local/bin/dhcp_relay.sh start (code=exited, status=0/SUCCESS)
    Process: 288891 ExecStart=/usr/local/bin/dhcp_relay.sh wait (code=exited, status=0/SUCCESS)
    Process: 288897 ExecStop=/usr/local/bin/dhcp_relay.sh stop (code=exited, status=0/SUCCESS)
   Main PID: 288891 (code=exited, status=0/SUCCESS)
[20:58:24.307889412] RUNNING: sudo systemctl reset-failed dhcp_relay at
[20:58:24.334046824] RUNNING: systemctl status dhcp_relay at
● dhcp_relay.service - dhcp_relay container
     Loaded: loaded (/lib/systemd/system/dhcp_relay.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/dhcp_relay.service.d
             └─auto_restart.conf
     Active: inactive (dead) since Tue 2022-01-18 20:58:24 UTC; 62ms ago
    Process: 288791 ExecStartPre=/usr/local/bin/dhcp_relay.sh start (code=exited, status=0/SUCCESS)
    Process: 288891 ExecStart=/usr/local/bin/dhcp_relay.sh wait (code=exited, status=0/SUCCESS)
    Process: 288897 ExecStop=/usr/local/bin/dhcp_relay.sh stop (code=exited, status=0/SUCCESS)
   Main PID: 288891 (code=exited, status=0/SUCCESS)
[20:58:24.354766442] RUNNING: sudo systemctl restart dhcp_relay at
[20:58:27.154154336] RUNNING: systemctl status dhcp_relay at
● dhcp_relay.service - dhcp_relay container
     Loaded: loaded (/lib/systemd/system/dhcp_relay.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/dhcp_relay.service.d
             └─auto_restart.conf
     Active: active (running) since Tue 2022-01-18 20:58:27 UTC; 10ms ago
    Process: 288999 ExecStartPre=/usr/local/bin/dhcp_relay.sh start (code=exited, status=0/SUCCESS)
   Main PID: 289086 (dhcp_relay.sh)
      Tasks: 2 (limit: 3303)
     Memory: 2.3M
     CGroup: /system.slice/dhcp_relay.service
             ├─289086 /bin/bash /usr/local/bin/dhcp_relay.sh wait
             └─289088 /usr/bin/python3 /usr/local/bin/sonic-db-cli STATE_DB hget WARM_RESTART_ENABLE_TABLE|system enable
admin@vlab-01:~$ bash script.bat dhcp_relay
```
* Run `bash script rsyslog.service`:
```
admin@vlab-01:~$ bash script rsyslog.service
[21:00:15.029910479] RUNNING: systemctl status rsyslog.service at
● rsyslog.service - System Logging Service
     Loaded: loaded (/lib/systemd/system/rsyslog.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2022-01-18 20:51:23 UTC; 8min ago
TriggeredBy: ● syslog.socket
       Docs: man:rsyslogd(8)
             man:rsyslog.conf(5)
             https://www.rsyslog.com/doc/
   Main PID: 285098 (rsyslogd)
      Tasks: 5 (limit: 3303)
     Memory: 1.5M
     CGroup: /system.slice/rsyslog.service
             └─285098 /usr/sbin/rsyslogd -n
[21:00:15.044423865] RUNNING: sudo systemctl restart rsyslog.service at
[21:00:15.148983057] RUNNING: sudo systemctl restart rsyslog.service at
[21:00:15.220294765] RUNNING: sudo systemctl restart rsyslog.service at
[21:00:15.279297717] RUNNING: sudo systemctl restart rsyslog.service at
[21:00:15.373179516] RUNNING: sudo systemctl restart rsyslog.service at
[21:00:15.439243686] RUNNING: sudo systemctl restart rsyslog.service at
Job for rsyslog.service failed.
See "systemctl status rsyslog.service" and "journalctl -xe" for details.
[21:00:15.501367333] RUNNING: systemctl status rsyslog.service at
● rsyslog.service - System Logging Service
     Loaded: loaded (/lib/systemd/system/rsyslog.service; enabled; vendor preset: enabled)
     Active: failed (Result: start-limit-hit) since Tue 2022-01-18 21:00:15 UTC; 51ms ago
TriggeredBy: ● syslog.socket
       Docs: man:rsyslogd(8)
             man:rsyslog.conf(5)
             https://www.rsyslog.com/doc/
    Process: 289449 ExecStart=/usr/sbin/rsyslogd -n (code=exited, status=0/SUCCESS)
   Main PID: 289449 (code=exited, status=0/SUCCESS)
[21:00:15.526606314] RUNNING: sudo systemctl reset-failed rsyslog.service at
[21:00:15.561726284] RUNNING: systemctl status rsyslog.service at
● rsyslog.service - System Logging Service
     Loaded: loaded (/lib/systemd/system/rsyslog.service; enabled; vendor preset: enabled)
     Active: inactive (dead) since Tue 2022-01-18 21:00:15 UTC; 112ms ago
TriggeredBy: ● syslog.socket
       Docs: man:rsyslogd(8)
             man:rsyslog.conf(5)
             https://www.rsyslog.com/doc/
    Process: 289449 ExecStart=/usr/sbin/rsyslogd -n (code=exited, status=0/SUCCESS)
   Main PID: 289449 (code=exited, status=0/SUCCESS)
[21:00:15.578779295] RUNNING: sudo systemctl restart rsyslog.service at
[21:00:15.658626654] RUNNING: systemctl status rsyslog.service at
● rsyslog.service - System Logging Service
     Loaded: loaded (/lib/systemd/system/rsyslog.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2022-01-18 21:00:15 UTC; 37ms ago
TriggeredBy: ● syslog.socket
       Docs: man:rsyslogd(8)
             man:rsyslog.conf(5)
             https://www.rsyslog.com/doc/
   Main PID: 289476 (rsyslogd)
      Tasks: 5 (limit: 3303)
     Memory: 2.9M
     CGroup: /system.slice/rsyslog.service
             └─289476 /usr/sbin/rsyslogd -n
admin@vlab-01:~$ 
```